metrics.py

```diff
@@ -88,16 +97,61 @@ def calc_change_score(self, net_1: np.ndarray, net_2: np.ndarray, indices: tuple
         sign = 1 if net_2[indices[0], indices[1]] else -1
         return sign * np.dot(net_1[indices[0]], net_1[:, indices[1]])
 
+class BaseDegreeVector(Metric):
+    """
```

Comment:
It's really not important, but 2 thoughts about this:
1. Why is `calculate` implemented in the parent class and not in the In/Out classes themselves? There is no shared code in this method between the cases.
2. A degree is well defined for undirected graphs as well (and the calculation is identical to either of the calculations for the directed case). Maybe support this case as well (get the `is_directed` field as an argument to the constructor of `BaseDegreeVector`, and add another class that inherits from `BaseDegreeVector` and calls `super().__init__()` with `self.is_directed` to `False`).